src/proxy.rs

```diff
@@ -502,11 +527,129 @@ fn handle_server_event(
             );
 
             // TODO(#145): Fix unwrap
-            let handle = client_to_proxy.idle_reject(status).unwrap();
+            let handle = client_to_proxy.state.idle_reject(status).unwrap();
             trace!(role = "p2c", ?handle, "idle_reject");
         }
         client::Event::IdleDoneSent { handle } => {
             trace!(role = "p2s", ?handle, "--->");
         }
     }
 }
+
+struct ImapState<S> {
+    input_role: &'static str,
+    input_direction: &'static str,
+    input_color: Option<Color>,
+    input_fragmentizer: Fragmentizer,
+    output_role: &'static str,
+    output_direction: &'static str,
+    output_color: Option<Color>,
+    output_fragmentizer: Fragmentizer,
+    state: S,
+}
+
+impl ImapState<Client> {
+    pub fn client(state: Client) -> Self {
+        Self {
+            input_role: "s2p",
+            input_direction: "<--|",
+            input_color: Some(Color::Blue),
```

Comment:
Do you think we could simplify the `input_color` by not making it `Option`al?

Reply:
It was useful because I played a while with different colors. But if you like I can make `input_color` non-optional and remove `output_color`.

I added a commit. Resolve the conversation if you agree.